iib/workers/tasks/build.py

```diff
@@ -105,42 +105,54 @@ def _cleanup():
 
     :param int request_id: the ID of the IIB build request
     :param iter arches: an iterable of arches to create the manifest list for
+    :param build_tags: list of extra tag to use for intermediate index image
     :return: the pull specification of the manifest list
     :rtype: str
     :raises IIBError: if creating or pushing the manifest list fails
     """
     buildah_manifest_cmd = ['buildah', 'manifest']
-    output_pull_spec = get_rebuilt_image_pull_spec(request_id)
-    log.info('Creating the manifest list %s locally', output_pull_spec)
-    run_cmd(
-        buildah_manifest_cmd + ['create', output_pull_spec],
-        exc_msg=f'Failed to create the manifest list locally: {output_pull_spec}',
-    )
-    for arch in sorted(arches):
-        arch_pull_spec = _get_external_arch_pull_spec(request_id, arch, include_transport=True)
-        log.debug(
-            'Adding the manifest %s to the manifest list %s', arch_pull_spec, output_pull_spec
+    output_pull_specs = [get_rebuilt_image_pull_spec(request_id)]
+    _tags = [request_id]
+    if build_tags:
+        _tags += build_tags
+
+    for tag in _tags:
+        conf = get_worker_config()
+        output_pull_spec = conf['iib_image_push_template'].format(
+            registry=conf['iib_registry'], request_id=tag
         )
+        log.info('Creating the manifest list %s locally', output_pull_spec)
         run_cmd(
-            buildah_manifest_cmd + ['add', output_pull_spec, arch_pull_spec],
-            exc_msg=(
-                f'Failed to add {arch_pull_spec} to the' f' local manifest list: {output_pull_spec}'
-            ),
+            buildah_manifest_cmd + ['create', output_pull_spec],
+            exc_msg=f'Failed to create the manifest list locally: {output_pull_spec}',
         )
+        for arch in sorted(arches):
+            arch_pull_spec = _get_external_arch_pull_spec(request_id, arch, include_transport=True)
+            log.debug(
+                'Adding the manifest %s to the manifest list %s', arch_pull_spec, output_pull_spec
+            )
+            run_cmd(
+                buildah_manifest_cmd + ['add', output_pull_spec, arch_pull_spec],
+                exc_msg=(
+                    f'Failed to add {arch_pull_spec} to the'
+                    f' local manifest list: {output_pull_spec}'
+                ),
+            )
 
-    log.debug('Pushing manifest list %s', output_pull_spec)
-    run_cmd(
-        buildah_manifest_cmd + ['push', '--all', output_pull_spec, f'docker://{output_pull_spec}'],
-        exc_msg=f'Failed to push the manifest list to {output_pull_spec}',
-    )
+        log.debug('Pushing manifest list %s', output_pull_spec)
+        run_cmd(
+            buildah_manifest_cmd
+            + ['push', '--all', output_pull_spec, f'docker://{output_pull_spec}'],
+            exc_msg=f'Failed to push the manifest list to {output_pull_spec}',
+        )
 
-    return output_pull_spec
+    return output_pull_specs[0]
```

Comment:
Could you add a comment on why is taken the first item of the list?

Reply:
You mean in the comment or here? First item is taken as first item is production tag, next items in the list are extra build tags

Reply:
Comment in code :)